R/content.R

```diff
@@ -1434,3 +1434,75 @@ get_content_packages <- function(content) {
   res <- content$packages()
   parse_connectapi_typed(res, connectapi_ptypes$content_packages)
 }
+
+# Integrations ----
+
+#' Set all OAuth integration associations for a content item
+#'
+#' @description
+#' Removes any existing OAuth integration associations for a content item, and
+#' creates associations with the integrations provided. You must have
+#' administrator or publisher privileges to perform this action.
+#'
+#' @param content A `Content` R6 object representing the content item to modify.
+#' @param integrations A single `connect_integration` object or a list of
+#'   `connect_integration` objects to associate with this content.
+#'
+#' @return Invisibly returns `NULL`. A message is printed on success.
+#'
+#' @seealso
+#' [get_integrations()], [integration()], [content_item()]
+#'
+#' @examples
+#' \dontrun{
+#' client <- connect()
+#'
+#' content <- content_item(client, "12345678-90ab-cdef-1234-567890abcdef")
+#'
+#' integrations <- get_integrations(client)
+#'
+#' # Associate a single integration
+#' github_integration <- purrr::keep(integrations, \(x) x$template == "github")[[1]]
+#' content_set_all_integrations(content, github_integration)
+#'
+#' # Associate multiple integrations at once
+#' selected_integrations <- integrations[1:2]
+#' content_set_all_integrations(content, selected_integrations)
+#' }
+#'
+#' @family oauth integration functions
+#' @family content functions
+#' @export
+content_set_all_integrations <- function(content, integrations) {
+  validate_R6_class(content, "Content")
+  # Handle a single integration
+  if (inherits(integrations, "connect_integration")) {
+    integrations <- list(integrations)
+  } else if (!inherits(integrations, "list")) {
+    stop(
+      "`integrations` must be a `connect_integration` class object or a list ",
+      "of `connect_integration` objects."
+    )
+  }
+  # Ensure that all the items we've been passed are integrations
+  if (!all(vapply(integrations, inherits, logical(1), "connect_integration"))) {
+    stop("All items must be `connect_integration` objects")
+  }
+
+  json_payload <- jsonlite::toJSON(
```

Comment:
The `PUT` method assumes json encoding, so is it necessary to `toJSON` here? That's not something we do anywhere else is it? If it is necessary here, it is worth a comment explaining.

Reply:
Yeah it's fine. Thanks for the pointer.